src/lib/layers/SpeedLimits.svelte

```diff
@@ -0,0 +1,94 @@
+<script lang="ts">
+  import { onMount, onDestroy } from "svelte";
+  import {
+    emptyGeojson,
+    overwriteSource,
+    overwriteLayer,
+    drawLine,
+  } from "../../maplibre_helpers";
+  import type { LineString } from "geojson";
+  import type {
+    DataDrivenPropertyValueSpecification,
+    GeoJSONSource,
+  } from "maplibre-gl";
+  import type { Feature } from "../../types";
+  import { lineWidth } from "../../colors";
+  import { map, gjScheme } from "../../stores";
+  import { type Remote } from "comlink";
+  import { type RouteInfo } from "../../worker";
+  import MapTooltips from "../common/MapTooltips.svelte";
+  import DiscreteLegend from "../common/DiscreteLegend.svelte";
+  import HelpIcon from "../common/HelpIcon.svelte";
+
+  export let routeInfo: Remote<RouteInfo>;
```

Comment:
Plumbing is getting kind of annoying. It's logically a global; maybe we should use a Svelte store.

Reply:
Feels like it should be a store? Especially if we're going to draw out more contextual layers like this.